state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -0,0 +1,256 @@
+    readings_correct_array = []
+    for acc in acc_readings:
+        xdir = abs(abs(numpy.dot(acc, numpy.array([1, 0, 0]))) - 9.8) < 1
+        ydir = abs(abs(numpy.dot(acc, numpy.array([0, 1, 0]))) - 9.8) < 1
+        zdir = abs(abs(numpy.dot(acc, numpy.array([0, 0, 1]))) - 9.8) < 1
+        readings_correct_array.append([xdir, ydir, zdir])
+
+    # if there is a True value in each subarray, find its index
+    # (x = 0, y = 1, z = 2)
+    correct_directions = []
+    for i in range(3):
+        correct_index = -1
+        if True in readings_correct_array[i]:
+            correct_index = readings_correct_array[i].index(True)
+        correct_directions.append(correct_index)
+
+    # populate xyz_correct and return
+    # if a direction's index is in correct_directions, we read g correct in
+    # that direction at least once
+    xyz_correct = [(i in correct_directions) for i in range(3)]
+    result = xyz_correct == [True, True, True]
+    return result, xyz_correct
+
+
+def gravity_imu_test(result_dict):
+    """
+    Check that we read g correctly in all 3 directions, given any user
+    input order
+    """
+    # get readings from all 3 sides
+    acc_readings = []
+    for i in range(3):
+        prompt = "Please leave the cubesat flat on one side."
+        if i > 0:
+            prompt = "Please leave the cubesat flat on another side."
+        res = request_imu_data(prompt)
+        if res is None:
+            result_dict["IMU_AccGravity"] = (
+                "Gravity test not completed.", None)
+            return result_dict
+        acc, _, _ = res
+        acc_readings.append(numpy.array(acc))
+
+    result, xyz_correct = check_gravity_acc(acc_readings)
+    xdir, ydir, zdir = xyz_correct
+
+    if result:
+        result_string = f"""Accelerometer read g as approx. 9.8 m/s^2 in
+all 3 directions. x: {xdir}, y: {ydir}, z: {zdir}"""
+        print(result_string)
+        result_dict["IMU_AccGravity"] = (result_string, True)
+    else:
+        result_string = f"""Failed to read g m/s^2 in all 3 directions.
+x: {xdir}, y: {ydir}, z: {zdir}"""
+        print(result_string)
+        result_dict["IMU_AccGravity"] = (result_string, False)
+    return result_dict
+
+
+def rotating_imu_test(result_dict):
+    """
+    Check that the norm of gyro is greater than 1
+    """
+    prompt = f"""Please rotate the cubesat for {wait_time} seconds
+once you start the test."""
+    res = request_imu_data(prompt)
+
+    if res is None:
+        result_dict["IMU_GyroRotating"] = (
+            "Rotating test not completed.", None)
+        return result_dict
+
+    _, gyro, _ = res
+    gyro_string = (f"Gyro: {tuple(gyro)} (deg/s)")
+    gyro_rotating = norm(gyro) >= 1
+
+    if gyro_rotating:
+        print(f"Gyro reading is too small with norm {norm(gyro)} ≤ 1 deg/s.")
+    else:
+        print("Rotating gyroscope reading is near 0 deg/s.")
+
+    result_dict["IMU_GyroRotating"] = (gyro_string, gyro_rotating)
+    return result_dict
```

Comment:
We don't need to return the result_dict, in any of the functions. Just a `return` should suffice.